Package.swift

```diff
@@ -15,7 +15,9 @@ let package = Package(
     .library(name: "AsyncSequenceValidation", targets: ["AsyncSequenceValidation"]),
     .library(name: "_CAsyncSequenceValidationSupport", type: .static, targets: ["AsyncSequenceValidation"])
   ],
-  dependencies: [],
+  dependencies: [
+    .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
```

Comment:
one idea that might be good here is that we conditionally do this by environment parameter; reducing the porting cost to embedded systems.

e.g. if we add the following to the top of the Package.swift
```
#if canImport(Darwin)
import Darwin
let buildingDocs = getenv("BUILDING_FOR_DOCUMENTATION_GENERATION") != nil
#elseif canImport(Glibc)
import Glibc
let buildingDocs = getenv("BUILDING_FOR_DOCUMENTATION_GENERATION") != nil
#else
let buildingDocs = false
#endif

// Only require the docc plugin when building documentation
let dependencies: [Package.Dependency] = buildingDocs ? [
  .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
] : []
```

and then change this to `dependencies: dependencies,`

and then `export BUILDING_FOR_DOCUMENTATION_GENERATION=1` from the doc build script

Reply:
```swift
#if canImport(Darwin)
import Darwin
let buildingDocs = getenv("BUILDING_FOR_DOCUMENTATION_GENERATION") != nil
#elseif canImport(Glibc)
import Glibc
let buildingDocs = getenv("BUILDING_FOR_DOCUMENTATION_GENERATION") != nil
```
Can this be simplified to be the following?
```swift
import Foundation
let buildingDocs = ProcessInfo.processInfo.environment["BUILDING_FOR_DOCUMENTATION_GENERATION"] == nil
```
One question is do we need to `#if canImport(Foundation)`. I’m not quiet familiar with the current’s state of swift for embedded systems support.

Reply:
Technically I also wanted this project to have a potential at offering bits to Foundation too; since the `AsyncBufferedByteIterator` is the backing implementation for `FileHandle.AsyncBytes`. Making it use just getenv poses a singular requirement on a common ancestor to all potential projects.